app/models/service_discovery/model_extensions.rb

```diff
@@ -22,35 +35,72 @@ def import_cluster_service_endpoint(cluster_service)
       end
 
       def import_cluster_active_docs(cluster_service)
-        unless cluster_service.oas?
-          log_cluster_service_import_event(cluster_service, message: 'API specification type not supported',
-                                                            details: { api_spec_content_type: cluster_service.specification_type })
-          return
-        end
+        return unless valid_cluster_service_spec?(cluster_service)
 
-        if (spec_content = cluster_service.specification).blank?
-          log_cluster_service_import_event(cluster_service, message: 'OAS specification is empty and cannot be imported')
-          return
-        end
-
-        api_docs_service = api_docs_services.build(name: cluster_service.name,
-                                                   body: spec_content,
-                                                   published: true,
-                                                   skip_swagger_validations: true)
+        api_docs_service = discovered_api_docs_service || build_api_doc_service(cluster_service)
+        api_docs_service.skip_swagger_validations = true
+        api_docs_service.body = cluster_service.specification_body
 
         unless api_docs_service.save
           log_cluster_service_import_event(cluster_service, message: 'Could not create ActiveDocs',
                                                             details: { errors: api_docs_service.errors.full_messages })
         end
       end
 
+      def discovered_api_docs_service
+        api_docs_services.discovered.first
+      end
+
       protected
 
+      def build_api_doc_service(cluster_service)
+        api_docs_services.build(name: cluster_service.name, published: true).tap { |object| object.discovered = true }
+      end
+
+      def valid_cluster_service_spec?(cluster_service)
```

Comment:
The methods having `cluster_service` does not belong to this class.

Reply:
It's because of the logs. The validation does belong to another class and it's there. This is just a wrapper to include the log messages in the context of a `Service` importing its new values and child objects (the `ApiDocs::Service` in this case).